src/twisted/conch/scripts/cftp.py

```diff
@@ -920,11 +921,44 @@ def _cbSubsystem(self, result):
                 f = open(fn)
         self.stdio = stdio.StandardIO(StdioClient(self.client, f))
 
-    def extReceived(self, t, data):
+    def extReceived(self, t: int, data: bytes) -> None:
         if t == connection.EXTENDED_DATA_STDERR:
             log.msg("got %s stderr data" % len(data))
-            sys.stderr.write(data)
-            sys.stderr.flush()
+            # RFC 4251
+            # ========
+            # Strings are also used to store text.  In that case, US-ASCII is
+            # used for internal names, and ISO-10646 UTF-8 for text that might
+            # be displayed to the user.  The terminating null character SHOULD
+            # NOT normally be stored in the string.  For example: the US-ASCII
+            # string "testing" is represented as 00 00 00 07 t e s t i n g.
+            # The UTF-8 mapping does not alter the encoding of US-ASCII
+            # characters.
+            #
+            # RFC 4254
+            # ========
+            # Additionally, some channels can transfer several types of data.  An
+            # example of this is stderr data from interactive sessions.  Such data
+            # can be passed with SSH_MSG_CHANNEL_EXTENDED_DATA messages, where a
+            # separate integer specifies the type of data.  The available types and
+            # their interpretation depend on the type of channel.
+            #
+            #    byte      SSH_MSG_CHANNEL_EXTENDED_DATA
+            #    uint32    recipient channel
+            #    uint32    data_type_code
+            #    string    data
+            #
+            # Data sent with these messages consumes the same window as ordinary
+            # data.
+            #
+            # Currently, only the following type is defined.  Note that the value
+            # for the 'data_type_code' is given in decimal format for readability,
+            # but the values are actually uint32 values.
+            #
+            #             Symbolic name                  data_type_code
+            #             -------------                  --------------
+            #           SSH_EXTENDED_DATA_STDERR               1
+            self.stderr.write(data.decode("utf-8", "backslashreplace"))
```

Comment:
Why is the choice to use a `backslashreplace` error handler? The standard makes no mention of error handling. Maybe it makes sense to error out by default, and have the developer handle the exception? Maybe pass in a specific error handler explicitly?

This is not a blocker. If it is not changed, maybe it's worthwhile explicitly documenting it as a design choice?

Reply:
I added some more comments about these choices - hopefully they make the motivation for the policy choice more clear (and maybe even guide a future maintainer to making a better choice).